cirq-core/cirq/circuits/circuit.py

```diff
@@ -2588,7 +2589,7 @@ def _apply_unitary_circuit(
     state: np.ndarray,
     qubits: Tuple['cirq.Qid', ...],
     dtype: Type[np.complexfloating],
-) -> np.ndarray:
+) -> Optional[np.ndarray]:
```

Comment:
Please revert, the return value on line 2627 can be only None if protocols.apply_unitaries is called with default=None; without search argument it raises TypeError.  https://github.com/quantumlib/Cirq/pull/5669

Reply:
Reverted.